test/specs/EventsSpec.js

```diff
@@ -21,325 +21,355 @@ describe("Event Tests", function() {
           touch.initEvent("touchend");
 
-          testSlider.on('slideStop', function() {
-            flag = true;
-          });
+          testSlider.on('slideStop', spy);
           testSlider.data('slider')._mouseup(mouse);
 
-          expect(flag).not.toBeTruthy();
+          window.setTimeout(function() {
+            expect(spy).not.toHaveBeenCalled();
+            done();
+          });
         });
       });
 
     });
 
     describe("Enabled/Disabled tests", function() {
-      it("'slideDisabled' event is triggered properly and can be binded to", function() {
-        testSlider.on('slideDisabled', function() {
-          flag = true;
-        });
+      var spy;
+
+      beforeEach(function() {
+        spy = jasmine.createSpy('spy');
+      });
+
+      it("'slideDisabled' event is triggered properly and can be binded to", function(done) {
+        testSlider.on('slideDisabled', spy);
         testSlider.slider('disable');
-        expect(flag).toBeTruthy();
+        window.setTimeout(function() {
+          expect(spy).toHaveBeenCalled();
+          done();
+        });
       });
 
-      it("'slideDisabled' event is triggered properly and can be binded to", function() {
-          testSlider.on('slideEnabled', function() {
-            flag = true;
-          });
+      it("'slideDisabled' event is triggered properly and can be binded to", function(done) {
+          testSlider.on('slideEnabled', spy);
           testSlider.slider('disable');
           testSlider.slider('enable');
-          expect(flag).toBeTruthy();
+          window.setTimeout(function() {
+            expect(spy).toHaveBeenCalled();
+            done();
+          });
       });
 
-      it("'change' event is triggered properly and can be binded to", function() {
-        testSlider.on('change', function() {
-          flag = true;
-        });
+      it("'change' event is triggered properly and can be binded to", function(done) {
+        testSlider.on('change', spy);
         testSlider.slider("setValue", 3, false, true);
-        expect(flag).toBeTruthy();
+        window.setTimeout(function() {
+          expect(spy).toHaveBeenCalled();
+          done();
+        });
       });
     });
 
   }); // End of JQuery version tests
 
   describe("CommonJS version", function() {
     describe("Event repetition tests", function() {
-      var testSlider, numTimesFired;
+      var testSlider, numTimesFired, spy;
 
       beforeEach(function() {
         testSlider = new Slider("#testSlider2");
         numTimesFired = 0;
+        spy = jasmine.createSpy('spy');
       });
 
       afterEach(function() {
         testSlider.destroy();
       });
 
-      it("'slide' event is triggered only once per slide action", function() {
-        testSlider.on('slide', function() {
-          numTimesFired++;
-        });
+      it("'slide' event is triggered only once per slide action", function(done) {
```

Comment:
This test makes sure that 'slide' event handler is not called more than once, hence the `numTimesFired` variable.
```js
it("'slide' event is triggered only once per slide action", function(done) {
    var obj = {
        addOne: function() {
          numTimesFired++;
        }
    };
    spyOn(obj, 'addOne').and.callThrough();

    testSlider.on('slide', obj.addOne);
    testSlider._mousemove(mouse);

    window.setTimeout(function() {
        expect(numTimesFired).toBe(1);
        expect(obj.addOne.calls.count()).toBe(1);
        done();
    }, 0);
});
```